CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java

```diff
@@ -108,32 +122,62 @@ public AnalyticViewPane(final AnalyticController analyticController) {
+                            final ThreadConstraints localConstraints = ThreadConstraints.getConstraints();
+                            if (localConstraints.getCurrentReport() == null) {
+                                localConstraints.setCurrentReport(parentConstraints.getCurrentReport());
+                            }
+                            Platform.runLater(() -> {
+                                runButton.setText(RUN_STOP_TEXT);
+                                runButton.setStyle(RUN_STOP_STYLE);
+                            });
+
+                            running = true;
+                            try {
+                                final AnalyticQuestion<?> question = analyticConfigurationPane.answerCurrentQuestion();
+                                analyticResultsPane.displayResults(question);
+                            } catch (final AnalyticException ex) {
+                                LOGGER.log(Level.SEVERE, ex.getLocalizedMessage());
+                                final AnalyticQuestion<?> question = new AnalyticQuestion<>(analyticConfigurationPane.getCurrentQuestion());
+                                question.addException(ex);
+                                analyticResultsPane.displayResults(question);
+                            } finally {
+                                running = false;
+
+                                Platform.runLater(() -> {
+                                    runButton.setText(RUN_START_TEXT);
+                                    runButton.setStyle(RUN_START_STYLE);
+                                });
+                            }
+                        }, "Analytic View: Answer Question");
+                        questionThread.start();
+                        interaction.notify(PluginNotificationLevel.INFO, " * Working * ");
+                        questionThread.join();
+
+                        try {
+                            SwingUtilities.invokeAndWait(new Runnable(){
+                                @Override
+                                public void run() {
+                                    // waits for other queued tasks to complete, then does nothing and exits
+                                }
+                            });
+                        } catch (InvocationTargetException ex) {
+                            // Not severe enough to warrant an exception popup message to user
+                            Exceptions.printStackTrace(ex);
+                        }
                     }
-                }, "Analytic View: Answer Question");
-                questionThread = answerQuestionThread;
-                answerQuestionThread.start();
+                };
+
+                try {
+                    PluginExecution.withPlugin(virtualAnalytics).interactively(true).executeNow(activeGraph);
+                } catch (InterruptedException | PluginException ex) {
+                    LOGGER.log(Level.SEVERE, ex.getLocalizedMessage());
```

Comment:
Can you separate the `InterruptedException` and add a `Thread.currentThread().interrupt()`?